src/data/queries/alertSingle.ts

```diff
@@ -1,53 +1,69 @@
+import { QueryFormatter } from 'next-drupal-query'
+import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
 import {
-  QueryData,
-  QueryFormatter,
-  QueryParams,
-  QueryOpts,
-} from 'next-drupal-query'
-import { drupalClient } from '@/lib/drupal/drupalClient'
+  AlertSingle,
+  AlertBlock,
+  AlertNonReusable,
+} from '@/types/formatted/alert'
+import { formatParagraph } from '@/lib/drupal/paragraphs'
 import { queries } from '.'
-import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
-import { AlertSingle } from '@/types/formatted/alertSingle'
-import { transformAlertData } from '@/lib/drupal/query'
-
-export const params: QueryParams<null> = () => {
-  return queries
-    .getParams()
-    .addInclude([
-      'field_alert_block_reference',
-      'field_alert_block_reference.field_alert_content',
-      'field_alert_non_reusable_ref',
-      'field_alert_non_reusable_ref.field_va_paragraphs',
-    ])
-}
 
-type DataOpts = QueryOpts<null>
-
-export const data: QueryData<DataOpts, AlertSingle[]> = async (
-  opts
-): Promise<AlertSingle[]> => {
-  const entities = await drupalClient.getResourceCollection<AlertSingle[]>(
-    'paragraph--alert_single',
-    {
-      params: params().getQueryObject(),
-    }
-  )
-
-  return entities
-}
+// TODO:
+// We don't need this here:
+// export const params: QueryParams<null> = () => {
+//   return queries
+//     .getParams()
+//     .addInclude([
+//       'field_alert_block_reference',
+//       'field_alert_block_reference.field_alert_content',
+//       'field_alert_non_reusable_ref',
+//       'field_alert_non_reusable_ref.field_va_paragraphs',
+//     ])
+// }
+//
+// ...but it seems like we might want to define
+// paragraph-specific `includes` within the paragraph's
+// query file (this file), and then somehow reference these in
+// the node query when we need to include fields of a specific
+// paragraph type.
+//
+// E.g.
+// export const include = [
+//   'field_alert_block_reference',
+//   'field_alert_block_reference.field_alert_content',
+//   'field_alert_non_reusable_ref',
+//   'field_alert_non_reusable_ref.field_va_paragraphs',
+// ]
+//
+// /* Some util file*/
+// export const getIncludedSubFields = (
```

Comment:
Thanks! It seemed like it would work, but I didn't want to move (in this PR) beyond just noting this because, as mentioned, this is already big enough. I'm going to keep it as a comment and then perhaps move on to it soon/later, or leave it as an exercise for the reader ;)